Sources/HipMobileUI/ViewModels/Views/RegisterScreenViewModel.cs

```diff
@@ -23,125 +23,120 @@
-		}
-	}
+    public class RegisterScreenViewModel : NavigationViewModel
+    {
+        private String email;
+        private String password;
+        private String repassword;
+        private String errorMessage;
+        private readonly MainPageViewModel mainPageViewModel;
+
+
+        public RegisterScreenViewModel(MainPageViewModel mainPageVm)
+        {
+            mainPageViewModel = mainPageVm;
+            Register = new Command(OnRegisterClicked);
+
+        }
+
+        public ICommand Register { get; }
+
+
+        async void RegisterUser()
+        {
+            User user = new User(Email, password);
+            UserStatus userStatus = await IoCManager.Resolve<IUserManager>().Register(user);
+            if (userStatus == UserStatus.Registered)
+            {
+                mainPageViewModel.SwitchToLoginView();
+                await Application.Current.MainPage.DisplayAlert(Strings.RegisterScreenView_Alert_Registered, Strings.RegisterScreenView_Alert_Description, Strings.RegisterScreenView_Alert_Ok);
+            }
+            else
+                DisplayRegisterFailedErrorMessage();
+        }
+
+        void DisplayEmptyEmailAndPasswordErrorMessage()
+        {
+            ErrorMessage = Strings.RegisterScreenView_Error_Empty_Email_And_Password;
+        }
+
+        void DisplayRegisterFailedErrorMessage()
+        {
+            ErrorMessage = Strings.RegisterScreenView_Error_Register_Fail;
+        }
+
+        void DisplayEmptyPasswordErrorMessage()
+        {
+            ErrorMessage = Strings.RegisterScreenView_Error_Empty_Password;
+        }
+        void DisplayPasswordMismatchErrorMessage()
+        {
+            ErrorMessage = Strings.RegisterScreenView_Error_Mismatch_Password;
+        }
+        void DisplayEmptyEmailErrorMessage()
+        {
+            ErrorMessage = Strings.RegisterScreenView_Error_Empty_Email;
+        }
+        void DisplayInvalidEmailErrorMessage()
+        {
+            ErrorMessage = Strings.RegisterScreenView_Error_Invalid_Email;
+        }
+
+        void OnRegisterClicked()
+        {
+            const string emailRegex = @"^(?("")("".+?(?<!\\)""@)|(([0-9a-z]((\.(?!\.))|[-!#\$%&'\*\+/=\?\^`\{\}\|~\w])*)(?<=[0-9a-z])@))" +
+                @"(?(\[)(\[(\d{1,3}\.){3}\d{1,3}\])|(([0-9a-z][-\w]*[0-9a-z]*\.)+[a-z0-9][\-a-z0-9]{0,22}[a-z0-9]))$";
+
+            bool isValid = false;
+
+            if (Email != null)
+            {
+                isValid = (Regex.IsMatch(Email, emailRegex, RegexOptions.IgnoreCase));
+            }
+
+            if (String.IsNullOrWhiteSpace(Email) && (String.IsNullOrWhiteSpace(Password) || String.IsNullOrWhiteSpace(RepeatPassword)))
+
+                DisplayEmptyEmailAndPasswordErrorMessage();
+
+            else if (String.IsNullOrWhiteSpace(Email))
+
+                DisplayEmptyEmailErrorMessage();
+
+            else if (String.IsNullOrWhiteSpace(Password) || String.IsNullOrWhiteSpace(RepeatPassword))
```

Comment:
I would separate both cases for more precise user information.

Reply:
This has nothing to do with changing the new registering workflow. We could open a new issue/bug for this.